src/compiler/transformers/module/module.ts

```diff
@@ -1786,7 +1807,12 @@ namespace ts {
         scoped: true,
         text: `
             function __export(m) {
-                for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
+                for (var p in m) if (!exports.hasOwnProperty(p)) Object.defineProperty(exports, p, {
+                  enumerable: true,
+                  get: function () {
+                    return m[p];

```

Comment:
this won't work as expected. since `var p` is not lexically scoped, `p` will always be the last key when any getter is executed

Reply:
Ah. Good catch.  Do the helpers need to be es5, or is `let` viable here?

Reply:
I don't know for sure, but I think the helpers are inserted as is. That means they are not downleveled if necessary.
Instead you could use an IIFE and pass the key as parameter.

Reply:
Alright. I've fixed this using a function to scope the key.  I didn't use an iife so that the function is only created once per call to the helper.  I could have hoisted it up to the top level, but I didn't want to pollute the module scope any more than necessary.